zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterContainer.java

```diff
@@ -67,14 +73,36 @@ void initContainer(
     }
   }
 
-  void initPosition() {
+  private void initPosition() {
     position = exportersState.getPosition(getId());
     lastUnacknowledgedPosition = position;
     if (position == ExportersState.VALUE_NOT_FOUND) {
       exportersState.setPosition(getId(), -1L);
     }
   }
 
+  void initMetadata() {
+    final var curMetadata = exportersState.getMetadataVersion(getId());
+    final long metadataVersion = initializationInfo.metadataVersion();
+    if (metadataVersion <= curMetadata) {
+      // metadata is already up-to-date. Just update the in-memory position
+      initPosition();
```

Comment:
:upside_down_face: The way I read it, if the current metadata is older, then we want to do some extra set up. But at any rate, we always want to then do the normal initialization (e.g. `initPosition`). To future proof a little, I would then flip the condition and do the extra set up (maybe extracted to a different method, e.g. `initExporterState`) in the if block, omit the return, and then perform the rest of the set up. e.g.

```java
void initExporterState() {
  final var initializeFrom = initializationInfo.initializeFrom();
  if (initializeFrom != null) {
    final var metadata = exportersState.getExporterMetadata(initializeFrom);
    final var otherPosition = exportersState.getPosition(initializeFrom);
    exportersState.initializeExporterState(getId(), otherPosition, metadata, metadataVersion);
  } else {
    // No need to initialize metadata, just set the version and initialize the position
    exportersState.initializeExporterState(getId(), -1L, null, metadataVersion);
  }
}

void initMetadata() {
  final var curMetadata = exportersState.getMetadataVersion(getId());
  final var metadataVersion = initializationInfo.metadataVersion();

  if (metadataVersion > curMetadata) {
    initExporterState();
  }

  initPosition();
}
```